### PR TITLE
Jane Street packages v0.16.0

### PR DESCRIPTION
**Pull request by dkalinichenko-js**
This PR adds Jane Street packages v0.16.0.

### CODE REVIEW

File: packages/torch/torch.v0.16.0/opam [hunk 1]
```diff
@@ -0,0 +1,38 @@
+opam-version: "2.0"
```
Comment (kit-ty-kate) [inline packages/torch/torch.v0.16.0/opam:1]: @LaurentMazare Hi sorry to bother you. Could you confirm ocaml-torch is changing maintainer? I'm not seeing any mentioned of this change in the original repository https://github.com/LaurentMazare/ocaml-torch but janestreet maintainers told me the change was agreed.

Comment (LaurentMazare) [reply packages/torch/torch.v0.16.0/opam:1]: @kit-ty-kate thanks for checking. Indeed this change was agreed and I can confirm that Jane Street will maintain the ocaml-torch package going forward, I'll tweak the readme of my repo to make this explicit.

**Comment by avsm** [comment]
I'm closing this PR temporarily until the issues with it crashing the CI are resolved, as it's blocking all the others being tested. Will re-open as soon as that is sorted.  It is easier if these PRs come in as smaller chunks (e.g a new release of Base and its dependents, followed by Core, followed by everything else)

**Comment by tmcgilchrist** [comment]
This can be re-opened when you're ready @dkalinichenko-js @avsm

**Comment by mseri** [comment]
Reopening as requested. Let's see 🤞

**Comment by emillon** [comment]
If this crashes again can we wait until #23814 is merged?

**Comment by avsm** [comment]
I'm closing this again while #23814 is also completely hosing the cluster.  And @dkalinichenko-js might, I please get a view on my request earlier about getting this PR broken up into stages, as we used to do in earlier releases?

>  It is easier if these PRs come in as smaller chunks (e.g a new release of Base and its dependents, followed by Core, followed by everything else)

It's far easier to review and merge if we add upper bounds first for Base, and then add Core, and then the remaining libraries.

**Comment by dkalinichenko-js** [comment]
> I'm closing this again while #23814 is also completely hosing the cluster. And @dkalinichenko-js might, I please get a view on my request earlier about getting this PR broken up into stages, as we used to do in earlier releases?
> 
> > It is easier if these PRs come in as smaller chunks (e.g a new release of Base and its dependents, followed by Core, followed by everything else)
> 
> It's far easier to review and merge if we add upper bounds first for Base, and then add Core, and then the remaining libraries.

Hi! I opened https://github.com/ocaml/opam-repository/pull/23834 for the latest release of Base (and Sexplib0). It seems even draft PRs trigger the CI, so I'm going to avoid creating a PR for Core & dependencies for now. The branch for Core is located at https://github.com/janestreet/main-opam-repository/tree/janestreet-v0.16-core.

**Pull request closed without merging**